<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AESCryptography</class>
 <widget class="QWidget" name="AESCryptography">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>360</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QStackedWidget" name="stackedWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>620</width>
     <height>300</height>
    </rect>
   </property>
   <widget class="QWidget" name="page_about">
    <widget class="QTextBrowser" name="textBrowser_AESCryptographyAbout">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>620</width>
       <height>300</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'JetBrainsMono Nerd Font'; font-size:11pt; font-weight:500; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:20pt; font-weight:700; text-decoration: underline;&quot;&gt;AES Cryptography&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Page&lt;/span&gt;: AESCryptography -&amp;gt; About.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Description&lt;/span&gt;: UI and functions for encrypting text with AES, hashing with SHA256, and decrypting.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:20pt; font-weight:700; text-decoration: underline;&quot;&gt;Pages&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Encrypt&lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;: &lt;/span&gt;Encrypt and decrypt text securely using AES-256.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Hash&lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;: Hash text with SHA-256.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Convert&lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;: Convert between plain text (UTF-8), hex and bytes.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-weight:400;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:20pt; font-weight:700; text-decoration: underline;&quot;&gt;Encryption&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Encrypts plaintext using &lt;span style=&quot; font-weight:700;&quot;&gt;AES-256 (Advanced Encryption Standard) in GCM (Galois/Counter Mode)&lt;/span&gt; with a derived key from a passphrase.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;A cryptographically secure random &lt;span style=&quot; font-weight:700;&quot;&gt;salt&lt;/span&gt; and &lt;span style=&quot; font-weight:700;&quot;&gt;initialization vector (IV) &lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;are generated&lt;/span&gt;. A 256-bit encryption key is derived from the supplied passphrase using &lt;span style=&quot; font-weight:700;&quot;&gt;PBKDF2 with HMAC-SHA256&lt;/span&gt;. Encryption is performed using OpenSSLâ€™s EVP interface. An authentication tag is generated. The resulting output is a Base64-encoded byte sequence containing, in order: a format version byte, the salt, IV, GCM authentication tag, and the ciphertext.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Salt&lt;/span&gt;: (Pseudo)Random value added to the hash to prevent pre-computed dictionaries.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;IV&lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;:&lt;/span&gt; (Pseudo)Random value to ensure that the encryption of data is unique and secure, even if the same plaintext is encrypted multiple times.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;PBKDF2&lt;/span&gt;: Password-Based Key Derivation Function (Version 2). Takes the passphrase and salt, and uses a pseudorandom function (in this case HMAC) on them, with many interations (in this case 10_000_000). This derives a key.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;HMAC-SHA256&lt;/span&gt;: Hash-based Message Authentication Code using SHA-256 as the hashing algorithm. Combines the salt and passphrase before hashing.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;SHA256&lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;: Secure Hashing Algorith that outputs 256 bytes. O&lt;/span&gt;ne-way function that is nearly impossible to reverse-engineer. Takes in data and outputs something that is 256 bytes long.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;AES-256&lt;/span&gt;: Advanced encryption standard. Symmetric encryption algorithm, which encodes data with a key and plain text, and can decode data if the correct key and correct encoded data (cipher text) is provided. Originally called Rijndael. 256 means it uses a key with a length of 256 bytes.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;GCM&lt;/span&gt;&lt;span style=&quot; font-weight:400;&quot;&gt;: Galois/Counter Mode. For symmetric key cryptographic block ciphers. Takes in the key, cipher text and IV. Computes an authentication tag which can be used to check if the cipher text and associated data are not tampered with.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_hash">
    <widget class="QWidget" name="horizontalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>611</width>
       <height>261</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="hashHLayout">
      <item>
       <layout class="QVBoxLayout" name="vLayoutHashPlainText">
        <item>
         <widget class="QLabel" name="label_hashPlainText">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Plain Text</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="hashPlainTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonHash">
          <property name="text">
           <string>Hash</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::Scanner"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vLayoutHashedText">
        <item>
         <widget class="QLabel" name="label_HashedText">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Hashed Text (as hex)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="hashedTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="plainText">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonCopyHashedText">
          <property name="text">
           <string>Copy Hashed Text</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditCopy"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_hashTitle">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>620</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>Hash (With SHA-256)</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_convert">
    <widget class="QLabel" name="label_convertTitle">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>620</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>Convert</string>
     </property>
    </widget>
    <widget class="QWidget" name="horizontalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>788</width>
       <height>261</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="convertHLayout">
      <item>
       <layout class="QVBoxLayout" name="vLayoutConvertPlainText">
        <item>
         <widget class="QLabel" name="label_convertPlainText">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Plain Text</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="convertPlainTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonConvertPlainText">
          <property name="text">
           <string>Convert Plain Text</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::MailReplyAll"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonConvertCopyPlainText">
          <property name="text">
           <string>Copy Plain Text</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditCopy"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vLayoutBytes">
        <item>
         <widget class="QLabel" name="label_convertBytes">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Bytes</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="bytesTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonConvertBytes">
          <property name="text">
           <string>Convert Bytes</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::MailReplyAll"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonConvertCopyBytes">
          <property name="text">
           <string>Copy Bytes</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditCopy"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vLayoutHex">
        <item>
         <widget class="QLabel" name="label_convertHex">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Hex</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="hexTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonConvertHex">
          <property name="text">
           <string>Convert Hex</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::MailReplyAll"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonConvertCopyHex">
          <property name="text">
           <string>Copy Hex</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditCopy"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="page_encrypt">
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>611</width>
       <height>221</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="encryptHLayout">
      <item>
       <layout class="QVBoxLayout" name="vLayoutPlainText">
        <item>
         <widget class="QLabel" name="label_encryptPlainText">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Plain Text</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="encryptPlainTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="placeholderText">
           <string>Hello</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonEncrypt">
          <property name="text">
           <string>Encrypt</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::SystemLockScreen"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonEncryptCopyPlainText">
          <property name="text">
           <string>Copy Plain Text</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditCopy"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="vLayoutCipherText">
        <item>
         <widget class="QLabel" name="label_cipherText">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Cipher Text</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="cipherTextEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="placeholderText">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonDecrypt">
          <property name="text">
           <string>Decrypt</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditUndo"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="buttonCopyCipherText">
          <property name="text">
           <string>Copy Cipher Text</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::EditCopy"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_encryptTitle">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>620</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>Encrypt</string>
     </property>
    </widget>
    <widget class="QWidget" name="horizontalLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>270</y>
       <width>611</width>
       <height>31</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_encryptPassphrase">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>Passphrase:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="encryptPassphrase">
        <property name="placeholderText">
         <string>correct horse battery staple</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_encryptPassphrase_2">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>Interations:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="encryptIterations">
        <property name="maximumSize">
         <size>
          <width>120</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="placeholderText">
         <string notr="true">1000000</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
